component/local/file/file.go

```diff
@@ -55,20 +54,20 @@ var DefaultArguments = Arguments{
 	PollFrequency: time.Minute,
 }
 
-var _ gohcl.Decoder = (*Arguments)(nil)
+var _ river.Unmarshaler = (*Arguments)(nil)
 
-// DecodeHCL implements gohcl.Decoder.
-func (a *Arguments) DecodeHCL(body hcl.Body, ctx *hcl.EvalContext) error {
+// UnmarshalRiver implements river.Unmarshaler.
+func (a *Arguments) UnmarshalRiver(f func(interface{}) error) error {
 	*a = DefaultArguments
 
 	type arguments Arguments
-	return gohcl.DecodeBody(body, ctx, (*arguments)(a))
+	return f((*arguments)(a))
 }
 
 // Exports holds values which are exported by the local.file component.
 type Exports struct {
 	// Content of the file.
-	Content *hcltypes.OptionalSecret `hcl:"content,attr"`
+	Content rivertypes.OptionalSecret `river:"content,attr"`
```

Comment:
Was it intentional to change from a pointer?

Reply:
not intentional, but I think a non-pointer is correct actually (go-cty might've required a pointer, I can't remember)